src/Microsoft.AspNet.Mvc.Core/ModelBinding/TypeConverterModelBinder.cs

```diff
@@ -34,12 +36,18 @@ public sealed class TypeConverterModelBinder : IModelBinder
                    bindingContext.ModelName,
                    bindingContext.ModelMetadata,
                    newModel);
+                var isModelSet = true;
 
-                return new ModelBindingResult(
-                    newModel,
-                    bindingContext.ModelName,
-                    isModelSet: true,
-                    validationNode: validationNode);
+                if (newModel == null && !AllowsNullValue(bindingContext.ModelType))
```

Comment:
This deserves a comment about what case it handles (type converter returns null instead of throwing).